src/decoder.rs

```diff
@@ -1100,8 +1124,13 @@ impl Decoder {
                     let (va, next_input) = a.parse(program, scope, input)?;
                     input = next_input;
                     v.push(va);
-                    let vs = Value::Seq(v.clone());
-                    let done = expr.eval_lambda(scope, vs).unwrap_bool();
+                    let vs = Value::Seq(v);
+                    let done = expr.eval_lambda(scope, &vs).unwrap_bool();
+                    v = if let Value::Seq(v) = vs {
+                        v
+                    } else {
+                        panic!("wut");
+                    };
```

Comment:
Is there a nicer way of doing this?

Reply:
Yes!

```suggestion
let vs = Value::Seq(v);
let done = expr.eval_lambda(scope, &vs).unwrap_bool();
let Value::Seq(v) = vs else { unreachable!("wut") };
```

or even

```suggestion
let done = expr.eval_lambda(scope, &Value::Seq(v)).unwrap_bool();
```

Not sure if the second will complain about lifetimes but the first definitely should work.

Reply:
I think this let-else introduces a _new_ v, it doesn't assign to the existing variable.

Reply:
I changed it to a match and it looks neat enough.